File: 73-set-matrix-zeroes/73-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& grid) {
        
        int n=grid.size();
        int m=grid[0].size();
        
        vector<int> rows(n, 1), cols(m, 1);
        
        for(int i=0; i<n; i++){
            for(int j=0; j<m; j++){
                
                if(grid[i][j] == 0){
                    rows[i]=0;
                    cols[j]=0;
                }
            }
        }
        
        for(int i=0; i<n; i++){
            for(int j=0; j<m; j++){
                if(rows[i]==0 or cols[j]==0){
                    grid[i][j]=0;
                }
            }
        }
    }
};

File: 73-set-matrix-zeroes/73-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        
        ROWS, COLS = len(matrix), len(matrix[0])
        
        rows = [1]*ROWS
        cols = [1]*COLS
        
        for i in range(ROWS):
            for j in range(COLS):
                if matrix[i][j] == 0:
                    rows[i] = 0
                    cols[j] = 0
                    
        
        for i in range(ROWS):
            for j in range(COLS):
                if rows[i] == 0 or cols[j] == 0:
                    matrix[i][j] = 0
        
        
        